flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/ReduceITCase.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.streaming.runtime;
+
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ExecutionOptions;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.datastream.KeyedStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.test.util.AbstractTestBase;
+import org.apache.flink.util.CloseableIterator;
+import org.apache.flink.util.CollectionUtil;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link KeyedStream#reduce(ReduceFunction)}.
+ */
+public class ReduceITCase extends AbstractTestBase {
+	@Test
+	public void testStreamReduce() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(1);
+		Configuration conf = new Configuration();
+		conf.set(ExecutionOptions.RUNTIME_MODE, RuntimeExecutionMode.STREAMING);
+		env.configure(conf, this.getClass().getClassLoader());
+		DataStreamSource<Long> numbers = env
+			.fromSequence(0, 10);
+
+		// send all records into a single reducer
+		KeyedStream<Long, Long> stream = numbers.keyBy(i -> i % 2);
+		DataStream<Long> sums = stream.reduce(
+			Long::sum
+		);
+
+		try (CloseableIterator<Long> sumsIterator = sums.executeAndCollect()) {
+			List<Long> results = CollectionUtil.iteratorToList(sumsIterator);
+			assertThat(results, equalTo(Arrays.asList(
+				0L, 1L, 2L, 4L, 6L, 9L, 12L, 16L, 20L, 25L, 30L
+			)));
+		}
+	}
+
+	@Test
+	public void testBatchReduce() throws Exception {
```

Comment:
Maybe we should add this to the `DataStreamBatchExecutionITCase` that I'm introducing in #13891. But we should at least not choose `ReduceITCase` because there's already too many classes with that name... 😅

Reply:
Sure, will move the tests into the `DataStreamBatchExecutionITCase`, once we get it in.